golem-worker-executor-base/src/durable_host/wasm_rpc/mod.rs

```diff
@@ -426,40 +430,89 @@ impl<Ctx: WorkerCtx> HostWasmRpc for DurableWorkerCtx<Ctx> {
+
+            Self::add_self_parameter_if_needed(&mut function_params, payload);
+
+            let current_idempotency_key = self
+                .state
+                .get_current_idempotency_key()
+                .expect("Expected to get an idempotency key as we are inside an invocation");
+
+            let current_oplog_index = self.state.current_oplog_index().await;
+
+            let idempotency_key =
+                IdempotencyKey::derived(&current_idempotency_key, current_oplog_index);
+
+            let serializable_input = SerializableScheduleInvocationRequest {
+                remote_worker_id: remote_worker_id.worker_id(),
+                idempotency_key: idempotency_key.clone(),
+                function_name: function_name.clone(),
+                function_params: try_get_typed_parameters(
+                    self.state.component_service.clone(),
+                    &remote_worker_id.account_id,
+                    &remote_worker_id.worker_id.component_id,
+                    &function_name,
+                    &function_params,
+                )
+                .await,
+                datetime: <SerializableDateTime as From<DateTime<Utc>>>::from(datetime.into()),
+            };
 
-        let current_oplog_index = self.state.current_oplog_index().await;
+            let action = ScheduledAction::Invoke {
+                owned_worker_id: remote_worker_id,
+                idempotency_key,
+                full_function_name: function_name,
+                function_input: function_params.into_iter().map(|e| e.into()).collect(),
+            };
 
-        let idempotency_key =
-            IdempotencyKey::derived(&current_idempotency_key, current_oplog_index);
+            let result = self
+                .state
+                .scheduler_service
+                .schedule(datetime.into(), action)
+                .await;
+
+            durability
+                .persist_serializable(self, serializable_input, Ok(result.clone()))
+                .await?;
 
-        let action = ScheduledAction::Invoke {
-            owned_worker_id: remote_worker_id,
-            idempotency_key,
-            full_function_name,
-            function_input: function_input.into_iter().map(|e| e.into()).collect(),
+            result
+        } else {
+            durability.replay::<ScheduleId, GolemError>(self).await?
         };
 
-        // The scheduler uses the (deterministic) datetime to deduplicate scheduled actions.
-        // The worker executor uses the idempotency key to deduplicate invocation.
-        // With both together we don't need any extra idempotency handling here.
-        self.state
-            .scheduler_service
-            .schedule(datetime.into(), action)
-            .await;
+        let cancellation_token = CancellationTokenEntry {
+            schedule_id: bincode::encode_to_vec(schedule_id, bincode::config::standard()).unwrap(),
```

Comment:
should not we use the `serialize` function from golem-common? (which prefixes the bincode binary with a serialization version number, to allow us switch serialization formats in the future)

Reply:
Makes sense, will update when back

Reply:
done